devops/scripts/deploy-public-upstream.sh

```diff
@@ -56,15 +56,17 @@ done
 
 git commit -F /tmp/commit_message --author='Pantheon Automation <bot@getpantheon.com>'
 
-# update the release-pointer
-git tag -f -m 'Last commit set on upstream repo' release-pointer HEAD
-
 # Push released commits to a few branches on the upstream repo.
 # Since all commits to this repo are automated, it shouldn't hurt to put them on both branch names.
 release_branches=('master' 'main')
 for branch in "${release_branches[@]}"; do
   git push public public:"$branch"
 done
 
+git checkout $CIRCLE_BRANCH
+
+# update the release-pointer
+git tag -f -m 'Last commit set on upstream repo' release-pointer HEAD
```

Comment:
Right now, newcommits is calculated by looking at all commits in `release-pointer..HEAD`. `release-pointer` is tagged against the "upstream" repository, but pushed to origin. So the release-pointer commit in origin is only referenced from this one tag, and does not exist on any branch. The commit that is tagged (prior to this PR) is whatever is at HEAD of the public branch, which is master/main of the target repository.

I guess that $CIRCLE_BRANCH here is always the `release` branch? If that's the case, then it seems that this fix is an improvement; when the next release comes around, we want to find all of the newer commits on the release branch. It seems there wouldn't necessarily be a relationship between those commits and the old HEAD of the public branch. I'm wondering why the existing code apparently works, though.

Reply:
@alexdicianu Could you please review the proposed change and comment on how the existing code works?

Reply:
I think the existing code doesn't work but it hasn't really been tested in drupal-project. It worked the first time that a squashing commit was done, then the next time was a commit from Tom to both repos and nothing else has entered to the upstream. I tried with current code and it didn't work because the diff will always include some commits already squashed because the hashes will be different